tools/rule_extractor/rule_extractor.py

```diff
@@ -0,0 +1,147 @@
+    if not route:
+        sys.stderr.write("Expected endpoint route but got None")
+        sys.exit(1)
+
+    endpoint = "https://{}:{}/api/prometheus/api/v1/{}".format(ip, port, route)
+    try:
+        response = requests.get(url=endpoint, verify=False)
+    except requests.exceptions.ConnectionError as exc:
+        sys.stderr.write(
+            "Failed to reach endpoint {}: {!s}".format(route, exc)
+        )
+        sys.exit(1)
+
+    if response.status_code != 200:
+        sys.stderr.write(
+            "Expected status code <200> got {}".format(response.status_code)
+        )
+        sys.exit(1)
+
+    return response.json()
+
+
+def write_to_file(content, rule_name):
+    filename = '{}.json'.format(rule_name)
+    filepath = (pathlib.Path(__file__)/'..'/'..'/'..'/'..'
+            /'tools'/'rule_extractor'/filename).resolve()
+    with open('{}.json'.format(rule_name), 'w', encoding='utf-8') as f:
+        f.write(json.dumps(content, indent=4, sort_keys=True))
+
+
+def main():
+    """
+        python rule_extractor --ip 127.0.0.1 --port 9090 --type rules
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-i',
+        '--ip',
+        dest='ip',
+        help='Specify the target control-plane ingress IP address'
+    )
+    parser.add_argument(
+        '-p',
+        '--port',
+        dest='port',
+        default='8443',
+        help='Specify the target control-plane ingress port',
+        type=int
+    )
+    parser.add_argument(
+        '-t',
+        '--type',
+        required=True,
+        dest='object_type',
+        help='Specify the type of objects to fetch',
+        choices=['rules', 'alerts']
+    )
+
+    args = parser.parse_args()
+    if not args.ip:
+        # Obtain VM ip by connecting to it and running a command
+        # Requires that the VM is up and running.
+        ip = subprocess.check_output(
+            [
+                "vagrant",
+                "ssh",
+                "--command",
+                "hostname -I | cut -d' ' -f1"
+            ]
+        ).decode(sys.stdout.encoding).strip()
+        # Validate that we got a real ipv4 address
+        try:
+            ipaddress.ip_address(ip)
+        except ValueError:
+            sys.stderr.write("Ip address: {} is invalid".format(ip))
+            sys.exit(1)
+    else:
+        ip = args.ip
+        try:
+            ipaddress.ip_address(ip)
+        except ValueError:
+            sys.stderr.write("Ip address: {} is invalid".format(ip))
+            sys.exit(1)
+
+    if args.object_type == "rules":
+        result = query_prometheus_api(ip=ip, port=args.port, route="rules")
+        # For documentation purposes, we don't want to litter the whole thing
+        # with all fields, so lets strip off some "ambiguous" fields
+        # Also we need to test that this generated default alerting rules are
+        # actually same as the deployed alerting rules
```

Comment:
I don't understand the purpose of this comment, it's not related to this script, no tests here.